core/trino-main/src/main/java/io/trino/event/SplitMonitor.java

```diff
@@ -31,21 +30,26 @@
 import java.time.Duration;
 import java.util.Optional;
 
+import static java.lang.Math.toIntExact;
 import static java.time.Duration.ofMillis;
 import static java.util.Objects.requireNonNull;
 
 public class SplitMonitor
 {
     private static final Logger log = Logger.get(SplitMonitor.class);
 
-    private final ObjectMapper objectMapper;
+    private final JsonCodec<DriverStats> driverStatsJsonCodec;
     private final EventListenerManager eventListenerManager;
+    private final int maxJsonLimit;
 
     @Inject
-    public SplitMonitor(EventListenerManager eventListenerManager, ObjectMapper objectMapper)
+    public SplitMonitor(EventListenerManager eventListenerManager, JsonCodec<DriverStats> driverStatsJsonCodec,
+            QueryMonitorConfig queryMonitorConfig)
     {
         this.eventListenerManager = requireNonNull(eventListenerManager, "eventListenerManager is null");
-        this.objectMapper = requireNonNull(objectMapper, "objectMapper is null");
+        this.driverStatsJsonCodec = requireNonNull(driverStatsJsonCodec, "objectMapper is null");
+        this.maxJsonLimit = toIntExact(
+                requireNonNull(queryMonitorConfig, "queryMonitorConfig is null").getMaxOutputStageJsonSize().toBytes());
```

Comment:
given the naming of the config is tied to query monitor, I think we can create a new config for split monitor. People may value tracking/storage of query and split stats at different granularity - so having separately configurable length is desirable imo.